src/Microsoft.DotNet.VersionTools/Util/FileUtils.cs

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.DotNet.VersionTools.Util
+{
+    static class FileUtils
+    {
+        public static bool ReplaceFileContents(string path, Func<string, string> replacement)
+        {
+            string contents;
+            Encoding encoding;
+
+            // Atttempt to preserve the file's encoding, using ASCII encoding (which has no BOM) if the file's
+            // encoding cannot be detected. 
+            using (StreamReader reader = new StreamReader(new FileStream(path, FileMode.Open), Encoding.ASCII, true))
```

Comment:
If the file does not have a BOM, then the StreamReader cannot detect its encoding, so it uses the default encoding.  Thus, we want the default encoding to correspond to no BOM, so that's why ASCII is the default here.

If UTF-8 were the default here, then it would add the UTF-8 BOM when the file originally had no BOM, which is current behavior that this PR is fixing.